pkg/tabulator/tabstate.go

```diff
@@ -245,3 +260,46 @@ func TabStatePath(configPath gcs.Path, tabPrefix, dashboardName, tabName string)
 	}
 	return np, nil
 }
+
+func calculateState(ctx context.Context, client gcs.Client, log *logrus.Entry, fromPath, toPath gcs.Path, confirm bool) error {
```

Comment:
nit: calculateState doesn't indicate to me that this does uploads, is there a better name to clarify that?

Reply:
Chose a more vague name, since this function currently "does everything" for a particular tab.